pkg/pillar/utils/generics/lockmap.go

```diff
@@ -75,3 +75,26 @@ func (lm *LockedMap[K, V]) Range(callback LockedMapFunc[K, V]) {
 	}
 	lm.RUnlock()
 }
+
+// LockedMapApplyFunc is function signature for ApplyOrStore function
+type LockedMapApplyFunc[V any] func(V) V
+
+// ApplyOrStore is used to perform atomic operations, where the operation
+// is performed by the applyFn function. If the key does not exist in the map
+// then defaultVal will be stored. Otherwise the entry for the key will be updated.
+// returns true if value exists
```

Comment:
nit: if the key (not value) existed (when the function was called) since the key always exists on return.

Reply:
updated text :)